mne/viz/ica.py

```diff
@@ -83,10 +84,11 @@ def plot_ica_sources(ica, inst, picks=None, exclude=None, start=None,
     from ..evoked import Evoked
     from ..epochs import BaseEpochs
 
-    if exclude is None:
-        exclude = ica.exclude
-    elif len(ica.exclude) > 0:
-        exclude = np.union1d(ica.exclude, exclude)
+    if exclude != 'deprecated':
+        warn('The ``exclude`` parameter is deprecated and will be removed in '
+             'version 0.20; specify excluded components using the '
+             '``ICA.exclude`` attribute instead.', DeprecationWarning)
+    exclude = ica.exclude
```

Comment:
this is a behavior change. I think it should be:
```
    if exclude != 'deprecated':
        warn(...)
    else:
        exclude = None
    # retain old logic
    if exclude is None:
        exclude = ica.exclude
    elif len(ica.exclude) > 0:
        exclude = np.union1d(ica.exclude, exclude)
```

Reply:
that was intentional.  My thinking was: fix the bug (side effect) immediately, and do a deprecation cycle for the API change (removing the parameter).  Knowing that, do you still think it should go back to the old behavior?

Reply:
At the very least the warning message should say that the behavior has changed and that this parameter is already ignored completely in the current release. This is different from how we usually deprecate things, but I can live with it as a bug fix. In that case it should also have an entry in the `BUG` section, as that's where people should be able to look for "behaviors that change immediately without a deprecation cycle".